library/std/src/sys/uefi/alloc.rs

```diff
@@ -1,22 +1,37 @@
 //! Global Allocator for UEFI.
 //! Uses [r-efi-alloc](https://crates.io/crates/r-efi-alloc)
 
-use crate::alloc::{GlobalAlloc, Layout, System};
+use r_efi::protocols::loaded_image;
 
-const MEMORY_TYPE: u32 = r_efi::efi::LOADER_DATA;
+use crate::alloc::{GlobalAlloc, Layout, System};
+use crate::sync::OnceLock;
+use crate::sys::uefi::helpers;
 
 #[stable(feature = "alloc_system_type", since = "1.28.0")]
 unsafe impl GlobalAlloc for System {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        static MEMORY_TYPE: OnceLock<u32> = OnceLock::new();
```

Comment:
This might be clearer (or at least easier to reference) if it use the specced enum name.

```suggestion
        static EFI_MEMORY_TYPE: OnceLock<u32> = OnceLock::new();
```